lib/protoparser/zabbixconnector/parser.go

```diff
@@ -0,0 +1,267 @@
+package zabbixconnector
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+	"github.com/VictoriaMetrics/metrics"
+	"github.com/valyala/fastjson"
+)
+
+const (
+	ADDGROUPS    byte = 0b00000001
+	ADDEMPTYTAGS byte = 0b00000010
+	EMPTYVALUE   byte = 49 // "1"
+)
+
+type Rows struct {
+	Rows []Row
+}
+
+func (rs *Rows) Reset() {
+	for i := range rs.Rows {
+		rs.Rows[i].reset()
+	}
+	rs.Rows = rs.Rows[:0]
+}
+
+func (rs *Rows) Unmarshal(s string, par byte) {
+	rs.Rows = unmarshalRows(rs.Rows[:0], s, par)
+}
+
+// Row is a single Zabbix Connector row
+type Row struct {
+	Tags      []Tag
+	Value     float64
+	Timestamp int64
+}
+
+func (r *Row) reset() {
+	tags := r.Tags
+	for i := range tags {
+		tags[i].reset()
+	}
+	r.Tags = tags[:0]
+
+	r.Value = 0
+	r.Timestamp = 0
+}
+
+func (r *Row) addTag() *Tag {
+	dst := r.Tags
+	if cap(dst) > len(dst) {
+		dst = dst[:len(dst)+1]
+	} else {
+		dst = append(dst, Tag{})
+	}
+	tag := &dst[len(dst)-1]
+	r.Tags = dst
+	return tag
+}
+
+func (r *Row) unmarshal(o *fastjson.Value, par byte) error {
+	r.reset()
+
+	host := o.GetObject("host")
+	if host == nil {
+		return fmt.Errorf("missing `host` object")
+	}
+
+	v := host.Get("host").GetStringBytes()
+	if len(v) == 0 {
+		return fmt.Errorf("missing `host` element in `host` object")
+	}
+	tag := r.addTag()
+	tag.Key = append(tag.Key[:0], []byte("host")...)
```

Comment:
Usually, in Prometheus like systems this label should be `instance`.
See [this](https://prometheus.io/docs/concepts/jobs_instances/) for example.

```suggestion
	tag.Key = append(tag.Key[:0], []byte("instance")...)
```

Also, for Prometheus-like systems there should be a label named `job` but I don't see anything similar in Zabbix. Am I missing something that can be used in similar manner or there is no similar label?

Reply:
In Zabbix, "host" refers to any physical or virtual device, application, service, or any other logically related set of monitored parameters.

I would say that a "host" in Zabbix can sometimes be thought of as a "job" in Prometheus. But in Zabbix there is no direct correspondence with job and instance.

"host" in Zabbix is a technical name. It is mainly used by active agents to identify themselves in the monitoring system. As a rule, this is the server hostname, but can be overridden in the agent config. (For example: we use the unique node ID from the CMDB)
"hostname" - visible host name (alias). If you set this name, it will be the one visible in lists, maps, filters , rules etc instead of the technical host name.

https://www.zabbix.com/documentation/current/en/manual/config/hosts

I think these label names should be left as they are.